python_modules/libraries/dagster-dg-cli/dagster_dg_cli/cli/scaffold/branch/diagnostics.py

```diff
@@ -0,0 +1,315 @@
+            correlation_id=self.correlation_id,
+            timestamp=datetime.now().isoformat(),
+            level=level,
+            category=category,
+            message=message,
+            data=data or {},
+        )
+
+        self.entries.append(entry)
+
+        # Stream entry to file immediately if file exists
+        if self._output_file and self._output_file.exists():
+            self._append_entry_to_file(entry)
+
+    def _append_entry_to_file(self, entry: DiagnosticsEntry) -> None:
+        """Append a single entry to the JSON file by rewriting it."""
+        if not self._output_file:
+            return
+
+        try:
+            # Read current content
+            current_content = json.loads(self._output_file.read_text())
+
+            # Add new entry
+            entry_dict = {
+                "correlation_id": entry.correlation_id,
+                "timestamp": entry.timestamp,
+                "level": entry.level,
+                "category": entry.category,
+                "message": entry.message,
+                "data": entry.data,
+            }
+            current_content["entries"].append(entry_dict)
+
+            # Write back to file
+            self._output_file.write_text(json.dumps(current_content, indent=2))
+        except (json.JSONDecodeError, KeyError):
+            # If file is corrupted, recreate it with current entries
+            self._recreate_file_with_entries()
+
+    def _recreate_file_with_entries(self) -> None:
+        """Recreate the diagnostics file with all current entries."""
+        if not self._output_file:
+            return
+
+        diagnostics_data = {
+            "correlation_id": self.correlation_id,
+            "session_start": self.entries[0].timestamp
+            if self.entries
+            else datetime.now().isoformat(),
+            "level": self.level,
+            "entries": [
+                {
+                    "correlation_id": entry.correlation_id,
+                    "timestamp": entry.timestamp,
+                    "level": entry.level,
+                    "category": entry.category,
+                    "message": entry.message,
+                    "data": entry.data,
+                }
+                for entry in self.entries
+            ],
+        }
+        self._output_file.write_text(json.dumps(diagnostics_data, indent=2))
+
+    def log_ai_interaction(self, interaction: AIInteraction) -> None:
+        """Log an AI interaction."""
+        if not self._should_log("info"):
+            return
+
+        self.log(
+            level="info",
+            category="ai_interaction",
+            message="Claude API interaction",
+            data={
+                "prompt_length": len(interaction.prompt),
+                "response_length": len(interaction.response),
+                "token_count": interaction.token_count,
+                "tools_used": interaction.tools_used,
+                "duration_ms": interaction.duration_ms,
```

Comment:
If we expect to be evolving these data structures might be better to include the "data" representation as a method on them rather than defining it here. That makes it easier to ensure new fields are included.